core/src/main/java/org/elasticsearch/action/support/tasks/TransportTasksAction.java

```diff
@@ -124,6 +126,22 @@ private NodeTasksResponse nodeOperation(NodeTaskRequest nodeTaskRequest) {
         return clusterState.nodes().resolveNodesIds(request.nodesIds());
     }
 
+    protected void processTasks(TasksRequest request, Consumer<OperationTask> operation) {
+        if (request.taskId() != BaseTasksRequest.ALL_TASKS) {
+            // we are only checking one task, we can optimize it
+            Task task = taskManager.getTask(request.taskId());
+            if (task != null && request.match(task)) {
```

Comment:
I think an exception would be the right thing todo we should never hide stuff like this. Maybe you can add a generic `ResourceNotFoundException` that we can later use in other places as well to  replace stuff like `AliasMissingExcepiton` etc. that has a `404` response code?